manifest-build-tools/application/release_to_atlas.py

```diff
@@ -0,0 +1,242 @@
+            "check_provider": lambda atlas_version, provider: "/".join([self.atlas_url, self.box, "version/{0}/provider/{1}".format(atlas_version, provider)]),
+            "create_provider": lambda atlas_version, provider: "/".join([self.atlas_url, self.box, "version/{0}/providers".format(atlas_version)]),
+            "upload_box": lambda atlas_version, provider: "/".join([self.atlas_url, self.box, "version/{0}/provider/{1}".format(atlas_version, provider), "upload"])
+        }
+        return purpose_handler[purpose](atlas_version, provider)
+
+def parse_args(args):
+    """
+    Parse script arguments.
+    :return: Parsed args for assignment
+    """
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('--build-directory',
+                        required=True,
+                        help="A directory where box files laid in",
+                        action='store')
+
+    parser.add_argument('--atlas-url',
+                        help="Base URL for Atlas, default: https://atlas.hashicorp.com/api/v1",
+                        action='store')
+
+    parser.add_argument('--atlas-username',
+                        help="The account name of atlas, default: rackhd",
+                        action='store')
+
+    parser.add_argument('--atlas-name',
+                        help="The repo name under a specific account of atlas, default: rackhd",
+                        action='store')
+
+    parser.add_argument('--atlas-token',
+                        help="atlas access token",
+                        action='store')
+
+    parser.add_argument('--atlas-version',
+                        help="atlas access token",
+                        action='store')
+
+    parser.add_argument('--is-release',
+                        help="if this is a step of rlease build",
+                        default=False,
+                        action='store')
+
+    parsed_args = parser.parse_args(args)
+    return parsed_args
+
+def upload_boxs(build_directory, atlas, is_release, atlas_version):
+    """
+    The function will walk through all sub-folder under $build_directory, and for every *.box found:
+        1. retrieve its version
+        2. upload to atlas with this version
+    NOTICE:
+        1. Box version is calculated from box file name.
+        2. Box provider is hardcoded and default to virtualbox currently.
+           If need to upload boxs of multi-provider, there should be a way
+           to pass the provider information to this script.
+    """
+
+    box_files = common.find_specify_type_files(build_directory, ".box", depth=1)
+    if len(box_files) == 0:
+        print "No box found under {0}".format(build_directory)
+
+    for full_file_path in box_files:
+        if not atlas_version:
+            if is_release:
+                # Box file name is like "rackhd-ubuntu-14.04-1.2.3-20161207024UTC.box"
+                # Extract 1.2.3-20161207024UTC only
```

Comment:
Not a long name, but just 0.12.07 for short.
The logic in this snippet is a little chaotic.
I'll modify this to:
* if is_release, atlas_version must be needed.
* if not release, will extract atlas version 0.12.07 from 1.2.3-20161207024UTC
* all other conditions will exit 1 with error.